pkg/operator/ceph/cluster/controller.go

```diff
@@ -242,6 +242,11 @@ func (r *ReconcileCephCluster) reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, cephCluster, errors.Wrap(err, "failed to get cephCluster")
 	}
 
+	// update observedGeneration local variable with current generation value,
+	// because generation can be changed before reconile got completed
+	// CR status will be updated at end of reconcile, so to reflect the reconcile has finished
+	observedGeneration := cephCluster.ObjectMeta.Generation
```

Comment:
Even though the cluster status update is more complicated for this reconcile, let's see if we can set the observed generation at the same time instead of separately. Let's discuss.